pkg/controller/broker/v2alpha3/activemqartemis/activemqartemis_reconciler.go

```diff
@@ -225,6 +233,28 @@ func (reconciler *ActiveMQArtemisReconciler) ProcessDeploymentPlan(customResourc
 	return reconciler.statefulSetUpdates
 }
 
+func (reconciler *ActiveMQArtemisReconciler) ProcessAddressSettings(customResource *brokerv2alpha3.ActiveMQArtemis, prevCustomResource *brokerv2alpha3.ActiveMQArtemis, client client.Client) bool {
+
+	var log = logf.Log.WithName("controller_v2alpha2activemqartemis")
+	log.Info("Process addresssettings")
+
+	if len(customResource.Spec.AddressSettings.AddressSetting) == 0 {
+		return false
+	}
+
+	//we need to compare old with new and update if they are different.
+	return compareAddressSettings(&prevCustomResource.Spec.AddressSettings, &customResource.Spec.AddressSettings)
+}
+
+//returns true if currentAddressSettings need update
+func compareAddressSettings(currentAddressSettings *brokerv2alpha3.AddressSettingsType, newAddressSettings *brokerv2alpha3.AddressSettingsType) bool {
+
+	if len((*currentAddressSettings).AddressSetting) != len((*newAddressSettings).AddressSetting) || *(*currentAddressSettings).ApplyRule != *(*newAddressSettings).ApplyRule || !config.IsEqual((*currentAddressSettings).AddressSetting, (*newAddressSettings).AddressSetting) {
```

Comment:
Just because you can doesn't mean you should. We should strive to write in a way that is immediately obvious without having to think about it.